### PR TITLE
DOCSP-43090 - Authentication

### PR DESCRIPTION
**Pull request by mongoKart**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-43090
Staging: https://deploy-preview-57--docs-c.netlify.app/security/authentication/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/security/authentication.txt [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+   :values: reference
+ 
+.. meta::
+   :keywords: authorize, ecs, amazon
+
+Overview
+--------
+
+This guide describes the mechanisms you can use in the {+driver-short+} to authenticate
+users.
```
Comment (rustagir) [inline source/security/authentication.txt:23]:
```suggestion
This guide describes the mechanisms you can use in the {+driver-short+} to authenticate
users to MongoDB.
```

File: source/security/authentication.txt [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+To authenticate with this mechanism, set the following connection options:
+
+- ``username``: The username to authenticate. Percent-encode this value before including
+  it in the connection URI.
+- ``password``: The password to authenticate. Percent-encode this value before including
+  it in the connection URI.
+- ``authSource``: The MongoDB database to authenticate against. By default,
+  the {+driver-short+} authenticates against the database in the connection
+  URI, if you include one. If you don't, it authenticates against the ``admin`` database.
```
Comment (rustagir) [inline source/security/authentication.txt:47]:
```suggestion
  URI, if you include one. If you don't specify an authentication database,
  it authenticates against the ``admin`` database.
```

File: source/security/authentication.txt [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+.. _c-scram-sha-256:
+
+SCRAM-SHA-256
+-------------
+
+SCRAM-SHA-256, as defined by `RFC 7677 <https://www.rfc-editor.org/rfc/rfc7677>`__,
+is the default authentication mechanism on MongoDB deployments
+running MongoDB v4.0 or later.
+
+To authenticate with this mechanism, set the following connection options:
```
Comment (rustagir) [inline source/security/authentication.txt:39]:
```suggestion
To authenticate by using SCRAM-SHA-256, set the following connection options:
```

File: source/security/authentication.txt (hunk 2 above)
Comment (rustagir) [reply source/security/authentication.txt:47]: applies to all sections

**Review by rustagir — APPROVED** [review]
lgtm, just left a few improving suggestions

File: source/security/authentication.txt (hunk 2 above)
Comment (mongoKart) [reply source/security/authentication.txt:47]: didn't find any more instances, but fixed the one here

**Review by rcsanchez97 — APPROVED** [review]
LGTM

**Pull request merged**